### PR TITLE
Manage tournament endpoint

### PR DESCRIPTION
**Pull request by EthanDelage**

### CODE REVIEW

**Review by tdameros — APPROVED** [review]

**Comment by aLeuleu** [comment]
You should ´git rebase --interactive ´ and squash the tests files into the commit where you created the endpoint
Im that way we can clearly see if there is a test missing or not

File: tournament/src/api/views/update_settings_views.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            if not valid_registration_deadline:
+                return JsonResponse(data={'errors': [registration_deadline_error]}, status=400)
+            tournament.registration_deadline = new_registration_deadline
+
+        new_is_private = body.get('is-private')
+        if new_is_private is not None:
+            valid_is_private, is_private_error = TournamentView.is_valid_private(new_is_private)
+            if not valid_is_private:
+                return JsonResponse(data={'errors': [is_private_error]}, status=400)
+            tournament.is_private = new_is_private
```
Comment (hferraud) [inline tournament/src/api/views/update_settings_views.py:81]: You could maybe split in different function to improve readability.

File: tournament/src/tournament/authenticate_request.py [hunk 2]
```diff
@@ -36,12 +37,14 @@ def decode_jwt(encoded_jwt, errors):
 def get_user(user_id, encoded_jwt, errors):
     headers = {'Authorization': encoded_jwt}
     try:
-        response = requests.get(f'http://user-management-nginx/user/{user_id}/', headers=headers)
+        response = requests.get(f'http://127.0.0.1:8001/user/{user_id}/', headers=headers)
```
Comment (hferraud) [inline tournament/src/tournament/authenticate_request.py:40]: Define the address in settings.

**Review by hferraud — CHANGES_REQUESTED** [review]

File: tournament/src/api/views/update_settings_views.py (hunk 1 above)
Comment (EthanDelage) [reply tournament/src/api/views/update_settings_views.py]: 2721d9976923a087d6cc45517054737efa9c357e

File: tournament/doc/tournament-api-documentation.md [hunk 3]
```diff
@@ -90,10 +153,34 @@ errors can be combined
...
+> }
+
+### Responses
+
+> | http code      | content-type       | response                               |
+> |----------------|--------------------|----------------------------------------|
+> | `200`          | `application/json` | `{"id": 1, "name": "Tournament", ...}` |
+> | `400` / `403`  | `application/json` | `{"errors": ["AAA", "BBB", "..."]}`    |
 
 </details>
```
[the remark this answers is not in the document]
Comment (EthanDelage) [reply tournament/doc/tournament-api-documentation.md]: 2721d9976923a087d6cc45517054737efa9c357e

[the remark this answers is not in the document]
Comment (EthanDelage) [reply tournament/doc/tournament-api-documentation.md]: 3aeac6e85b50846fcf7efe57701a1039ae9a62b2

File: tournament/src/api/views/update_settings_views.py (hunk 1 above)
Comment (EthanDelage) [reply tournament/src/api/views/update_settings_views.py]: 3aeac6e85b50846fcf7efe57701a1039ae9a62b2

File: tournament/src/tournament/authenticate_request.py (hunk 2 above)
Comment (EthanDelage) [reply tournament/src/tournament/authenticate_request.py]: 794ccceacaf81fb73290cac30f20b983921f4a69

File: tournament/src/api/error_message.py [hunk 4]
```diff
@@ -16,3 +16,8 @@
 
 MISSING_IS_PRIVATE = 'Missing is-private field'
 IS_PRIVATE_NOT_BOOL = 'Is private must be a boolean'
+
+MISSING_NICKNAME = 'Missing nickname field'
+NICKNAME_TOO_SHORT = f'nickname must contain at least {settings.MIN_NICKNAME_LENGTH} characters'
+NICKNAME_TOO_LONG = f'nickname must contain less than {settings.MAX_NICKNAME_LENGTH} characters'
+NICKNAME_INVALID_CHAR = 'nickname may only contain letters, numbers and spaces'
```
[the remark this answers is not in the document]
Comment (EthanDelage) [reply tournament/src/api/error_message.py]: 794ccceacaf81fb73290cac30f20b983921f4a69

**Review by hferraud — APPROVED** [review]

File: tournament/src/tournament/urls.py [hunk 5]
```diff
@@ -17,10 +17,7 @@
 from django.contrib import admin
 from django.urls import include, path
 
-from .views import index
-
 urlpatterns = [
     path('admin/', admin.site.urls),
-    path('', index),
     path('tournament/', include("api.urls"))
```
Comment (hferraud) [inline tournament/src/tournament/urls.py:22]: Use single quote.

**Review by V-Fries — APPROVED** [review]

**Pull request merged**